.github/workflows/gen-docs.yml

```diff
@@ -0,0 +1,105 @@
+name: Generate Documentation
+
+on:
+  release:
+    types: [published]
+    branches: "master"
+
+jobs:
+  # Split the generation of the sourcekitten json across 8 jobs to speed up the process.
+  # At the end of each job the results are uploaded as an artifact
+  build-json:
+    name: Build sourcekitten json
+    runs-on: macOS-latest
+    strategy:
+      matrix:
+        chunk: [0,1,2,3,4,5,6,7]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Install
+        run: brew install sourcekitten
+      - name: Build
+        run: ./scripts/build-json-doc.sh ${{ matrix.chunk }} 8
+      - name: Upload
+        uses: actions/upload-artifact@v1
+        with:
+          name: sourcekitten-${{ matrix.chunk }}
+          path: ./sourcekitten/
+
+  # This job requires the previous ones to have finished. It then downloads all the
+  # json data generated by the previous jobs before running jazzy
+  run-jazzy:
+    name: Run Jazzy
+    needs: build-json
+    runs-on: macOS-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install
+      run: |
+        gem install jazzy
+        npm install lodash fs globby
+    - name: Chunk 1
```

Comment:
Maybe name should be zero based as well, would make reading easier.

Reply:
not sure it makes a huge difference

Reply:
> How long would the gen-docs process take if it wouldn't be parallelized?

It takes 75-80 mins which is too long as the GITHUB_TOKEN expires after an hour. Unfortunately when we split across multiple instances we have to re-build the whole of aws-sdk-swift on each instance.

Reply:
> It takes 75-80 mins which is too long as the GITHUB_TOKEN expires after an hour. Unfortunately when we split across multiple instances we have to re-build the whole of aws-sdk-swift on each instance.

Maybe we should add this as a comment in the yaml file will make it much clearer for future us.

Reply:
done